src/transport.cpp

```diff
@@ -85,6 +86,53 @@ namespace rhost {
 
                 disconnect();
             }
+
+            void read_stream_to_message(int fdr, const std::string& message_name) {
+                char line[pipeSize];
+                size_t len = pipeSize;
+                ssize_t nread;
+
+                for(;;) {
+#ifdef _WIN32
+                    nread = _read(fdr, line, pipeSize);
```

Comment:
On the other hand, since `pipe` call is different anyway, it's probably not worthwhile.